### PR TITLE
Tests for scrabble-score's optional extensions

### PR DESCRIPTION
**Pull request by juleskers**
So I set myself the challenge of doing scrabble-score's extensions for `:double`- and `:triple`-scoring letters.
In the course of doing so, I wrote tests. I thought these may be useful to the source, so here is my offer to include them upstream.

Comments, suggestions, nits and other (constructive) criticism welcome; this is my first contribution to exercism.io, so I may need some guidance ;-)

### CODE REVIEW

**Comment by juleskers** [comment]
Aha, the Travis tests fail because my new tests fail.
It seems the reference implementation doesn't actually do the double and triple letter scoring.

**Comment by petertseng** [comment]
Oh yeah! and before I forget, I have to ask to ignore all but the first test, as I did in the other PR

**Comment by IanWhitney** [comment]
I think this is a good start. Thanks.

**Comment by juleskers** [comment]
Thanks for the feedback! I'll try to incorporate it!
The #[ignore] will be added, but probably only after I've passed the Travis round at least once.
(Or is travis smart enough to un-ignore the tests?)

I may be slow to respond, but that won't mean I've abandoned this, it's just that I'm doing this in the brief lost minutes during my commute :-)

I'm undecided on `:garbage`;
I was considering that `:doubleSomeword` would be the syntax for a double-scoring word; The spec leaves open how this would be expected.

**Comment by petertseng** [comment]
> (Or is travis smart enough to un-ignore the tests?)

Yes, it does. I should have explained it, my bad for not doing so!
You don't necessarily need to look at this script, but if you are curious, our Travis CI runs https://github.com/exercism/xrust/blob/master/_test/check-exercises.sh#L42, thus unignoring all tests.

> I was considering that `:doubleSomeword` would be the syntax for a double-scoring word; The spec leaves open how this would be expected.

Ah! This could be a fine idea. You are right about the spec.

As in the other PR, this could be a thing all languages can benefit from, so it could be worth opening an issue in x-common to discuss how the spec should be interpreted. And if we reach an agreement, a PR can be sent to x-common.

If agreement can't be reached in x-common, we could possibly clarify the specification for our track by adding a exercises/scrabble-score/HINTS.md - this is a special file that gets appended to the README.

**Comment by juleskers** [comment]
update!
* I've #[ignored] all newly added tests
* example code updated to pass all tests (locally ran check_exercises.sh, passed!)
* slight tweak of one test name

feel free to nitpick over variable names and comments! It's the example after all, so it should be clear!

I'm opening a discussion issue on x-common next.

File: exercises/scrabble-score/example.rs [hunk 1]
```diff
@@ -1,40 +1,35 @@
...
-    values.insert('w', 4);
-    values.insert('x', 8);
-    values.insert('y', 4);
-    values.insert('z', 10);
-    values
+/// Calculates the value of the current character.
+///
+/// The previous characters' value is required in case the current
+/// token is a multiplier, which affects the score of the previous char.
+fn char_value(the_char: char, prev_value: u16) -> u16 {
```
Comment (petertseng) [inline exercises/scrabble-score/example.rs]: I might also consider `c` here.

**Comment by IanWhitney** [comment]
a delimiter like `[]` works for me

- double `[]`
- triple `<>`

- aw[e]some
- aw<e>some
- [awesome]
- <awesome>
- <aw[e]some>
- etc.

**Comment by IanWhitney** [comment]
And I'm fine having tests in place to handle the "extra credit" problems, even if those tests aren't in canonical.

**Comment by juleskers** [comment]
Ah, delimiters! I like how simple the implementation could become:
(I wonder if that compiles...? Mutating the multiplyer inside the match AND reading it in very the same expression looks maybe a bit too compact to work. But maybe Rust lets us have nice things!)

`````rust
score += Match c {
   '[' => multiplyer *= 2; 0,
   ']' => multiplyer /= 2; 0,
   '<' => multiplyer *= 3; 0,
   '>' => multiplyer /= 3; 0,
   'a'|'e'|'u'.... => 1, 
   ... => 2,
   _ => 0 
} * multiplyer
`````

**Comment by IanWhitney** [comment]
If I'm reading this PR correctly, there are just a few small changes @petertseng mentioned in comments. Are those blocking merge? Or can we move on this PR without those changes?

**Comment by petertseng** [comment]
I had a different reading of this PR: I thought all of my comments are unimportant, and what we would want to discuss before merge is:

> In-line notation seem super awkward to me. Parsing seems fraught with problems, as you mention. I think a more robust structure is necessary.

And that whatever notation we use should be added to HINTS.md

**Comment by IanWhitney** [comment]
Ah, yes. I saw the comment approving the ideas of delimiters, but hadn't checked the code to see if delimiters had been implemented.

**Comment by juleskers** [comment]
After the discussion on the common track, I understood that the optional exercise was intentionally left vague. Figuring out a syntax was explicitly said to be part of the exercise.
That's why I haven't pursued this further.

**Comment by IanWhitney** [comment]
Ok. Sounds like we can close, then. Thanks for your work on this, @juleskers!

**Pull request closed without merging**